packages/@aws-cdk/aws-stepfunctions-tasks/test/stepfunctions/start-execution.test.ts

```diff
@@ -215,3 +215,53 @@ test('Execute State Machine - Wait For Task Token - Missing Task Token', () => {
     });
   }).toThrow('Task Token is required in `input` for callback. Use JsonPath.taskToken to set the token.');
 });
+
+test('Execute State Machine - Associate With Parent', () => {
+  const task = new StepFunctionsStartExecution(stack, 'ChildTask', {
+    stateMachine: child,
+    input: sfn.TaskInput.fromObject({
+      token: sfn.JsonPath.taskToken,
+    }),
+    associateWithParent: true,
+  });
+
+  new sfn.StateMachine(stack, 'ParentStateMachine', {
+    definition: task,
+  });
+
+  expect(stack.resolve(task.toStateJson())).toEqual({
```

Comment:
use `toMatchObject` and remove some of the entries below – we only care about the input